### PR TITLE
feat(dev): Add pull-lock to Palette Mobile

### PR DESCRIPTION
**Pull request by MrSltun**
This PR resolves [] 

### Description
This PR adds [pull-lock](https://github.com/orta/pull-lock) to be used with husky after pulling from `origin`
This change is done to run `yarn install` in case there have been some changes in `package.json` and prints a message if there are any native changes to run `yarn install:all`

### CODE REVIEW

File: ios/Podfile.lock [hunk 1]
```diff
@@ -1467,7 +1467,7 @@ SPEC CHECKSUMS:
   SDWebImage: a7f831e1a65eb5e285e3fb046a23fcfbf08e696d
   SDWebImageWebPCoder: 908b83b6adda48effe7667cd2b7f78c897e5111d
   SocketRocket: f32cd54efbe0f095c4d7594881e52619cfe80b17
-  Yoga: 6aaa3c45f0fda0c6929f855121622669dabe1aaf
```
Comment (MrSltun) [inline ios/Podfile.lock:1470]: @gkartalis I tried to update this package, but I still see this change, should I keep push it with this PR or remove it?

File: ios/Podfile.lock [hunk 2]
```diff
@@ -1467,7 +1467,7 @@ SPEC CHECKSUMS:
...
+  Yoga: 57d2ffe418d024d56f8b0047f335c677e4c4d9ac
```
Comment (gkartalis) [inline ios/Podfile.lock:1470]: 👁️👁️

Comment (gkartalis) [reply ios/Podfile.lock:1470]: is this actually needed? Could you run:
```sh
cd ios && bundle exec pod update Yoga --no-repo-update
```

to see if the diff persists? If it does feel free to keep it

**Review by gkartalis — APPROVED** [review]
noice

File: ios/Podfile.lock (hunk 1 above)
Comment (gkartalis) [reply ios/Podfile.lock:1470]: if you did try 
```sh
cd ios && bundle exec pod update Yoga --no-repo-update
``` 

then feel free to leave it here

File: ios/Podfile.lock (hunk 2 above)
Comment (MrSltun) [reply ios/Podfile.lock:1470]: that's exactly what I ran, and it still persisted 🤷

Comment (gkartalis) [reply ios/Podfile.lock:1470]: noice lets leave it here

**Comment by artsyit** [comment]
:rocket: PR was released in [`v13.2.36`](https://github.com/artsy/palette-mobile/releases/tag/v13.2.36) :rocket:

**Pull request merged**